test/Microsoft.Azure.Devices.Edge.Test/EndToEnd.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Test.Common;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using NUnit.Framework;
+    using Serilog;
+
+    public class EndToEnd
+    {
+        CancellationTokenSource cts;
+
+        [SetUp]
+        public void Setup() => this.cts = new CancellationTokenSource(Context.Current.TestTimeout);
+
+        [TearDown]
+        public void Teardown() => this.cts.Dispose();
+
+        [Test]
+        public async Task TempSensor()
+        {
+            string edgeAgent = Context.Current.EdgeAgent.Expect(() => new ArgumentException());
+            string edgeHub = Context.Current.EdgeHub.Expect(() => new ArgumentException());
+            string tempSensor = Context.Current.TempSensor.Expect(() => new ArgumentException());
+            Option<Uri> proxy = Context.Current.Proxy;
+
+            CancellationToken token = this.cts.Token;
+
+            string name = "temp sensor";
+            Log.Information("Running test '{Name}'", name);
+            await Profiler.Run(
+                async () =>
+                {
+                    var iotHub = new IotHub(
+                        Context.Current.ConnectionString,
```

Comment:
Can some of this common code be refactored out.

Reply:
Yes, the code in the tests themselves (as opposed to supporting library code) needs to become much simpler and more succinct. But I'm holding off on refactoring here until I have a broader representation of tests.